### ISSUE TITLE
Error in Read10X

### ISSUE THREAD

**Issue by fly4all** [satijalab/seurat#643]
I'm trying to work through the Seurat example. But it gave me an error on trying to use Read10X.
What does it mean by barcode file?

> list.files(data_dir)
[1] "pbmc3k_filtered_gene_bc_matrices.tar" "practice_seurat.R"                   
> pbmc.data <- Read10X(data.dir = data_dir)
Error in Read10X(data.dir = data_dir) : Barcode file missing

**Comment by fly4all** [comment]
Issue resolved. The directory should have contain unzipped files which includes the barcode file

**Comment by Sophia409** [comment]
![image](https://user-images.githubusercontent.com/30619010/50556713-a844c980-0d17-11e9-8930-46f9e8c88d9f.png)
I unzip the three files,but it still shows that Barcode file missing,do you know the reason?
> list.files(data_dir)
[1] "barcodes.tsv" "features.tsv" "matrix.mtx"  
> pbmc.data <- Read10X(data.dir = data_dir)
Error in Read10X(data.dir = data_dir) : Barcode file missing

**Comment by evolvedmicrobe** [comment]
There is an important difference between 10X data from cellranger 3.0 and previous versions.  Your data appears to be from 3.0 because it has a `features.tsv` file, in this case, you want to keep the file format gzipped so that they are read and input correctly, in version 2.0 these files were unzipped.  For more details on the current output format see: https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices

**Comment by Sophia409** [comment]
> There is an important difference between 10X data from cellranger 3.0 and previous versions. Your data appears to be from 3.0 because it has a `features.tsv` file, in this case, you want to keep the file format gzipped so that they are read and input correctly, in version 2.0 these files were unzipped. For more details on the current output format see: https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices



> There is an important difference between 10X data from cellranger 3.0 and previous versions. Your data appears to be from 3.0 because it has a `features.tsv` file, in this case, you want to keep the file format gzipped so that they are read and input correctly, in version 2.0 these files were unzipped. For more details on the current output format see: https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices
Thank you for your reply!

**Comment by ninggongzi** [comment]
> Why I could't remove the points in vlnplot? Can you help me analysis my code? VlnPlot(object = tc, features = c("Adnp2", "Tshz1", "Tshz2"),ncol=1,point.size.use = NA) Error in ExIPlot(object = object, type = "violin", features = features, : unused argument (point.size.use = NA) | | yuhong | | yuhongruoyi@163.com | 签名由网易邮箱大师定制 On 1/1/2019 05:20，Nigel Delaney<notifications@github.com> wrote： There is an important difference between 10X data from cellranger 3.0 and previous versions. Your data appears to be from 3.0 because it has a features.tsv file, in this case, you want to keep the file format gzipped so that they are read and input correctly, in version 2.0 these files were unzipped. For more details on the current output format see: https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.

try “point.size.use = 0”

**Comment by thaocad** [comment]
Hi,
I would like to use Read10X on mouse bone marrow data available in  https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE128423  

Yet I got this same error message: Error in Read10X(data_dir) : Barcode file missing

But when I checked, the barcode file was there:
> list.files(data_dir)
[1] "GSM3674240_b2.barcodes.tsv" "GSM3674240_b2.genes.tsv"
[3] "GSM3674240_b2.matrix.mtx"

I also checked in the paper, they used "Cellranger toolkit (version 2.0.1, 10X Genomics)". Is it the source of this problem? What should I do to resolve this?

I tried Read10X() with both gzipped and unzipped files, both did not work with same error. FYI, I am using Seurat 3.0

Thanks!
